assets/js/components/ClusterLink.jsx

```diff
@@ -2,19 +2,26 @@ import React from 'react';
 
 import { Link } from 'react-router-dom';
 
-const ClusterLink = ({ cluster, children }) => {
+const truncatedLength = 14; // Using 14 as it cuts the uuid format after the 2nd -
+export const getClusterName = (cluster) => {
+  return cluster?.name || cluster?.id.slice(0, truncatedLength) + '...';
+};
+
+const ClusterLink = ({ cluster }) => {
+  const clusterName = getClusterName(cluster);
+
```

Comment:
Instead of just cutting this way, an alternative approach is letting the browser deal with this through the `text-ellipsis` class, you wanna try?

https://tailwindcss.com/docs/text-overflow#ellipsis

Reply:
Updated here: https://github.com/trento-project/web/pull/497/commits/e61e6185658417160e0717736af312ea2991eb09